### ISSUE TITLE
BUG- Update required python version in Pipfile

### ISSUE THREAD

**Issue by anunayasri** [Avaiga/taipy#1151]
**Description**
Taipy required python version `>= 3.9` but Pipfile mentions required `python_version` as `3`. It should be `>=3.9`. 
I was setting up the codebase in my local and the setup broker due to this. I am trying to use pdm for managing dependencies. pipenv is too slow on my system.

**How to reproduce**

```shell
brew install pdm
cd $PROJECT_ROOT
pdm install
```

Error -
```
❯ pdm install
Lock file does not exist
Updating the lock file...
🔒 Lock failed
Unable to find a resolution because the following dependencies don't work on all Python versions in the
range of the project's `requires-python`: >=3.6.
  python>=3.7 (from <Candidate apispec@6.3.0 from https://pypi.org/simple/apispec/>)
A possible solution is to change the value of `requires-python` in pyproject.toml to >=3.7.
See /var/folders/d7/dtz_zy9d30l96qxg2qzvv5nm0000gn/T/pdm-lock-st285xoy.log for detailed debug log.
[ResolutionImpossible]: Unable to find a resolution
```

**Runtime environment**
Please specify relevant indications.
 - Taipy version (or branch name): `develop` branch
 - OS: macOS Monterey

**Acceptance Criteria**
- [ ] Ensure local setup works with the python_version change
- [ ] Ensure automated tests work fine with this change

**Comment by trgiangdo** [comment]
Pipenv doesn't support the `python_version = ">=3.9"`, so the only way we can allow taipy to be installed in a virtual environment with python >= 3.9 is to have the `python_version = "3"`.

Unless we move away from Pipenv, I don't think there is anything we can do about this

**Comment by Chirag8023** [comment]
!Assign me

and what vale we should set in the .toml file then ?

**Comment by FlorianJacta** [comment]
You are assigned to this issue now @Chirag8023, thank you for your contribution!

**Comment by anunayasri** [comment]
Hi @FlorianJacta . What is the final decision on dep management? Is the project moving away from Pipenv?

**Comment by Chirag8023** [comment]
I think i'm unable to understand and solve this issue and also i don't have mac environment and in codespaces the `pdm install` running sucessfully

**Comment by FabienLelaquais** [comment]
@anunayasri 
No, Taipy remains on pipenv for the time being.
However your point is relevant.
It should be fixed in the root's `pyproject.toml` file, where, in the "[project]" section, there should be an explicit:
    `requires-python = ">= 3.9"`

**Comment by FabienLelaquais** [comment]
Actually, I just realized that this setting is already there. So there's nothing more to do for this one.

**Comment by anunayasri** [comment]
Does that mean that a dev can use a dep manager like `pdm` instead of `pipenv` to install the dependencies? Asking since the pipenv install was super slow on my system and I had a hard time using pdm with setup.

**Comment by FabienLelaquais** [comment]
It is indeed. That would be the kind-of default/standard so far.
We try to make the package available for the many.
uv is also a realistic alternative...

**Comment by FlorianJacta** [comment]
It seems that this issue can be closed as the solution is already implemented.

**Comment by anunayasri** [comment]
@FabienLelaquais I have used `pdm` and `uv` for installing dependencies in my local. I have shared the steps in a discussion thread (https://github.com/Avaiga/taipy/discussions/1955).

Since you have mentioned 
> uv is also a realistic alternative

I would love to work on integrating `uv` in taipy. I know that this is a big change for the project. I would be happy to assist the core team on this change. Do you think a task can be created for this and assigned to me?

**Comment by FabienLelaquais** [comment]
> @FabienLelaquais I have used `pdm` and `uv` for installing dependencies in my local. I have shared the steps in a discussion thread (#1955).
> 
> Since you have mentioned
> 
> > uv is also a realistic alternative
> 
> I would love to work on integrating `uv` in taipy. I know that this is a big change for the project. I would be happy to assist the core team on this change. Do you think a task can be created for this and assigned to me?

@anunayasri thanks a lot for your proposal.
To be perfectly honest, I'm really not sure that most users are ready yet to move to another package manager that the 'standard' one. Taipy targets Enterprise-level applications that bring with them a lot of corporate development and deployment processes, where we have little wiggle room.
I look forward to seeing a path where different package managers can be used in parallel. And I suspect that the section-based approach that pyproject,yaml provides would let us do that properly.
The point is: we can not exclude the regular pip and pipenv, but we would be glad to support uv, pdm or others...

**Comment by anunayasri** [comment]
Hi @FabienLelaquais. Thanks for the explanation. I believe there are two different flows installed here.

1. For _users_ of lib `taipy`: How can users install taipy and its dependencies? They should be able to any package manager like pip, pipenv, poetry, uv etc. 
2. For _developers_ of `taipy`: How can taipy devs install the dependencies? 

I think your concerns are about (1). I agree with those concerns. I believe (2) is completely independent of (1).

I was proposing to specify taipy deps such that (2) can be used by most package managers specially the modern ones like uv which is quite fast. I believe specifying the deps in `pyproject.toml` is how it can be done. Most package managers will support. The current Pipfile format is specifically for `pipenv`. As a dev I am stuck with `pipenv` and can't use other package manager like `uv`.

In summary, even if we move deps from Pipfile to pyproject.toml (1) will remain unaffected. This is my understanding. Kindly let me know if I am missing something here.
